src/lib/SaferNames/NameCore.hs

```diff
@@ -115,7 +115,8 @@ data NameBinder (s::E)  -- static information for the name this binds (note
                 (l::S)  -- scope under the binder (`l` for "local")
   = UnsafeMakeBinder { nameBinderName :: Name s l }
 
-withFresh :: (InjectableE s, Typeable s, Distinct n)
+-- TODO Why did this want the constraints Typable s and InjectableE s?
```

Comment:
Good catch. withFresh used to need them because the constructor for Name used to require Typeable/Injectable constraints. I changed that in 097de2023402fc2ac3bf5f7fd3e98c182da5c9b1 but I missed updating this case.

Reply:
Perfect.  I was wondering, because Injectable looks like a pretty magic typeclass, and I didn't want to accidentally weaken the safety of the system by removing a semantic constraint.